byway/src/main/webapp/track-interests.js

```diff
@@ -59,6 +88,25 @@ function createButtonForPlace(place) {
   let button = document.createElement("button");
   button.innerText = place;
   button.addEventListener('click', () => updateStatus(place, button));
-  button.className = "btn";
+  button.className = "interestBtn";
   return button;
 }
+
+/**
+ * Gets all potential interests from server to load or sets the user's
+ * specific interests for their specific trip with tripId. userInterests
+ * may be any type that is convertible to an array via Array.from. It will
+ * be sent to the server as JSON in the post body.
+ * @param {String tripId} value for tripId
+ * @param {Array} [userInterests] interests selected by user
+ */
+function fetchPlaces(tripId, /* optional */ userInterests) {
+  const url = '/api/places?' + new URLSearchParams({tripId}).toString();
+  const fetchArgs = {method: 'GET'};
```

Comment:
Note that since you're now returning the `fetch` call in two separate places, you can just in-line the value of `fetchArgs` like, `fetch(url)` for the `GET` case (get is the default, so you don't have to set it), or `fetch(url, {method: POST, body: ...})` for the `POST` case.

Reply:
Fixed